### DISCUSSION TITLE
Momentum inlet type and function sized wont match.

### DISCUSSION THREAD

**Discussion by Spudman105** [idaholab/moose#27778] [Q&A Modules: Navier-Stokes] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A Navier Stokes is the most appropriate category for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (avoid screenshots if possible, triple back quotes before/after pasted text, etc)

### Question

I am working on using the Navier-stokesFV action module for wave movement using a variable velocity. However I am unable to get the momentum_inlet_type and momentum_inlet_funtion to match sizes. my exact error is 
"(Modules/NavierStokesFV/momentum_inlet_function):  Size (1) is not the same as the number of fixed-velocity and fixed-pressure entries in momentum_inlet_types (size 0)" I have set the type to be flux-velocity however it still says im using fixed velocity and pressure is the error?

'''
[Modules]
  [NavierStokesFV]
    # General params
    compressibility = 'incompressible'
    add_energy_equation = true

    # Material properties
    density = 'rho'
    dynamic_viscosity = 'mu'
    thermal_conductivity = 'k'
    specific_heat = 'cp'

    # Initial conditinons
    initial_velocity = '1 1 0'
    initial_pressure = 1.0
    initial_temperature = 0.0

    # "inlet" boundary conditions
    inlet_boundaries = '3'
    momentum_inlet_types = 'flux-velocity'
    momentum_inlet_function = '1 ((2.9683*10^3)*sin(t*((2*pi)/(1.2*10^-8))))' #'pressure_z'
    energy_inlet_types = 'fixed-temperature'
    energy_inlet_function = '0'

    #"wall" boundary conditions
    wall_boundaries = 'rmax'
    momentum_wall_types = 'noslip '
    energy_wall_types = 'fixed-temperature'
    #energy_wall_function = '0 0 1'

    #"outlet" boundary conditions
    pin_pressure = true
    pinned_pressure_type = average
    pinned_pressure_value = 0

    #dont have but could include temp params

    mass_advection_interpolation = 'average'
    momentum_advection_interpolation = 'average'
    energy_advection_interpolation = 'average'
  []
[]
'''

### Additional information

Mesh size and type:
 Reynolds number:
 Discretization (finite element CG/DG, finite volume, etc):
 Models (turbulence, porous media, etc):
 Solver method (fully coupled, segregated, multiapps, etc):
 Base input you started from:

**Comment by GiudGiud** [comment, marked as answer]
Hello

The flux-velocity inlet type is not a fixed velocity BC. You specify it with another parameter giving the fluxes instead

Are you trying to specify a fixed (varying in time) velocity? If so change the inlet type

Guillaume

**Reply by Spudman105** [reply]
Hi!

Thank you for your reply, I am trying to make a fixed velocity that is dependent on time. I assumed that a fixed velocity had to be non-changing.

thank you 
Ben
